tests/networkpolicy_test.go

```diff
@@ -226,6 +176,31 @@ func skipNetworkPolicyRunningOnKindInfra() {
 	}
 }
 
+func createVMICirros(virtClient kubecli.KubevirtClient, namespace string, labels map[string]string) *v1.VirtualMachineInstance {
+	var err error
+	vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
+	vmi.Namespace = namespace
+	vmi.Labels = labels
+	vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Create(vmi)
+	Expect(err).ToNot(HaveOccurred())
+
+	return vmi
+}
+
+func checkPingSuccess(fromVmi *v1.VirtualMachineInstance, toIp string) {
```

Comment:
I think the original name is more correct: `assertPingSucceed` as it is indeed asserting.
But I would argue that we do not need a helper in this case at all, avoiding the need to add an offset.

I also have a behavior question here: What happens if the first two attempts fail and the rest succeed, will that end with a return value of "0"/success?

Reply:
> I think the original name is more correct: `assertPingSucceed` as it is indeed asserting.
> But I would argue that we do not need a helper in this case at all, avoiding the need to add an offset.
> 

Thought about it as well, but since it repeats in two times, and because the next pending PR #4099 
will add logic, and to have a symmetric API between ping fail and ping success, decided to put a function
(i understand one can argue that each PR on its on, see please the other reasons).
Will change the name if needed either on this PR or in the follow one #4099 up to you.
 
> I also have a behavior question here: What happens if the first two attempts fail and the rest succeed, will that end with a return value of "0"/success?

no, because we dont have eventually and ping "-w 8" pings 8 seconds deadline and return success iff (not typo this time)
all successed.

Reply:
Stupid question, what about implementing it with Eventually Consistently so we don't depend on ping behaviour ? it will make it consistently with `checkPingFailure`

Reply:
already done, PTAL
fixed both threads

Reply:
I see this from the man:

> If  ping does not receive any reply packets at all it will exit with code 1. If a packet count and deadline are both specified, and fewer than count packets are received by the time the deadline has
> arrived, it will also exit with code 1.  On other error it exits with code 2. Otherwise it exits with code 0. This makes it possible to use the exit code to see if a host is alive or not.
> 

So if -c and -w are set, it should be the case. Can you test it?

Reply:
> I see this from the man:

i can but as Quique and me suggested, and since PingFromVMConsole is limited to 10 seconds atm,
and we need more than 10 seconds (because in order to see failures we saw 10 seconds were still flaky), 
i think we should use consistently (pushed the change if you can take a look),
but meanwhile trying manually what you said (i believe it works, so let me know if you prefer it over consistently, imo consistently is better)

Reply:
> I see this from the man:
> 
> > If  ping does not receive any reply packets at all it will exit with code 1. If a packet count and deadline are both specified, and fewer than count packets are received by the time the deadline has
> > arrived, it will also exit with code 1.  On other error it exits with code 2. Otherwise it exits with code 0. This makes it possible to use the exit code to see if a host is alive or not.
> 
> So if -c and -w are set, it should be the case. Can you test it?

tested, return 1 if both -w and -c are supplied, 0 if just -w is supplied and some packets success.

but as Quique and me suggested, and since PingFromVMConsole is limited to 10 seconds atm,
and we need more than 10 seconds (because in order to see failures we saw 10 seconds were still flaky),
i think we should use consistently (pushed the change if you can take a look).

wdyt ?

Reply:
1. `Failed` is a verb in the past tense (or an adjective which is not a fit here), I think you should use as the original `Fail`.
2. I do not mind if you use ping application or repeating the console command (although the later is slower), but I want to point out that expecting for the ping to always succeed from start does not cover all scenarios. it depends on the specific case, which in this case may be fine (you should check). I mean, if the policy was in advance permissive, ping should continue working as you just asserted, on the other hand, if the policy was not permissive, and it just "opened" it up, then it will first fails and later pass.

Reply:
fixed point (1) PTAL